br/pkg/backup/client.go

```diff
@@ -452,7 +452,14 @@ func (bc *Client) BackupRanges(
 			elctx := logutil.ContextWithField(ectx, logutil.RedactAny("range-sn", id))
 			err := bc.BackupRange(elctx, sk, ek, req, metaWriter, progressCallBack)
 			if err != nil {
-				return errors.Trace(err)
+				// The error due to context cancel, stack trace is meaningless, the stack shall be suspended (also clear)
+				if errors.Cause(err) == context.Canceled {
+					errors.SuspendStack(err)
+					return err
+				} else {
+					return errors.Trace(err)
+				}
```

Comment:
i feel like this is something that should be handled in `errors.Trace` not here 🤔 

or we should change some `zap.Error` to `log.ShortError`...

Reply:
errors.Trace comes from another remote package, put it into errors.Trace will has impact other repo, I guess.

Zap.Error and ShortError may not suitable for a real error, which we need call stack for investigation